src/controllers/likeController.js

```diff
@@ -0,0 +1,145 @@
+    if (created) {
+      return res
+        .status(201)
+        .json({
+          status: 201,
+          data:{
+          message: `You have disliked an accomodation ${accomodationId}`}
+        });
+    }
+    return res
+      .status(200)
+      .json({
+        status: 200,
+        data: {
+        message: `You have undisliked an accomodation ${accomodationId}`}
+      });
+  } catch (error) {
+    return ApplicationError.internalServerError(error.message, res);
+  }
+};
+const allLikes = async (req, res) => {
+  try {
+    const { accomodation_id: accomodationId } = req.params;
+    const accomodationExist = await Accomodation.findOne({ where: { id: accomodationId } });
+    if (!accomodationExist) {
+      return ApplicationError.notFoundError(
+        'This accomodation does not exist',
+        res
+      );
+    }
+    const accomodationLikes = await Like.findAll({   // numbers of likes of an accomodation accomodation
+      where: { accomodationId, likeStatus:true },
+        include: [
+          {
+            model: Users,
+            attributes: ['firstname','lastname',]
+          }
+        ],
+        attributes: { exclude: ["id", "accomodationId", "userId", "createdAt", "updatedAt", "likeStatus"]}
+    });
+    if (!accomodationLikes.length) {
+      return res
+        .status(200)
+        .json({ status: 200, data:{total_likes: 0,  message: `No likes for this accomodation` }});
+    }
+    return res
+      .status(200)
+      .json({
+        status: 200,
+        data:{
+        total_likes: accomodationLikes.length,
+        likedBy: accomodationLikes,
+        message: `You succesfully got likes of accomodation ${accomodationId}`}
+      });
+  } catch (error) {
+    return ApplicationError.internalServerError(error.message, res);
+  }
+};
+const allDisLikes = async (req, res) => {
+  try {
+    const { accomodation_id: accomodationId } = req.params;
+    const accomodationExist = await Accomodation.findOne({ where: { id: accomodationId } });
+    if (!accomodationExist) {
+      return ApplicationError.notFoundError(
+        'This accomodation does not exist',
+        res
+      );
+    }
+    const accomodationDisLikes = await Like.findAll({   // numbers of likes of an accomodation accomodation
+      where: { accomodationId, likeStatus: false },
+        include: [
+          {
+            model: Users,
+            attributes: ['firstname','lastname',]
+          }
+        ],
+        attributes: { exclude: ["id", "accomodationId", "userId", "createdAt", "updatedAt", "likeStatus"]}
+    });
+    if (!accomodationDisLikes.length) {
+      return res
+        .status(200)
+        .json({ status: 200,data:{total_likes: 0,  message: `No likes for this accomodation` }});
+    }
+    return res
+      .status(200)
+      .json({
+        status: 200,
+        data:{
+        total_likes: accomodationDisLikes.length,
+        likedBy: accomodationDisLikes,
```

Comment:
likedBy should be an array of objects, remove User property in the array to avoid to much of nesting